numba/tests/test_dyn_array.py

```diff
@@ -509,6 +509,20 @@ def pyfunc(x):
 
 class ConstructorBaseTest(NrtRefCtTest):
 
+    def test_check_zeros_prototype(self):
+        def func():
+            return np.zeros((2, 2), dtype=np.float)
+        cfunc = nrtjit(func)
+        expected = np.zeros((2, 2), dtype=np.float)
+        result = cfunc()
+        np.testing.assert_equal(result, expected)
+        # errors
+        with self.assertRaises(TypingError) as cm:
+            cfunc(-1)
+        self.assertEqual(str(cm.exception), "TypingError: dtype 'np.float' is "
```

Comment:
```suggestion
        self.assertEqual(str(cm.exception), "dtype 'np.float' is "
```
suggest just checking the message contents, the type is asserted in the `self.assertRaises(TypingError)`.